Sources/Datadog/DatadogInternal/DatadogCoreProtocol.swift

```diff
@@ -106,10 +106,17 @@ public protocol DatadogCoreProtocol: AnyObject {
     ///   - message: The message.
     ///   - fallback: The fallback closure to call when the message could not be
     ///               processed by any Features on the bus.
-    func send(message: FeatureMessage, else fallback: @escaping () -> Void)
+    func send(message: FeatureMessage, sender: DatadogCoreProtocol, else fallback: @escaping () -> Void)
+
+    /// Flushes all asynchronous operations in the SDK by **blocking the caller thread** till their completion.
+    func flush()
+
+    /// Awaits completion of all asynchronous operations and deinitializes this instance of the SDK.
+    /// It **blocks the caller thread**.
+    func flushAndTearDown()
```

Comment:
I don't think this should be part of the protocol, it's a specific behaviour that is not a requirement of a core. It can be exposed by specific implementation depending on its inner logic, that includes blocking the caller thread.

Reply:
Removing it from the protocol for now 👍, but we will need it in similar form or another because of our [integration with CI Visibility](https://docs.datadoghq.com/continuous_integration/tests/swift/?tab=swiftpackagemanager#rum-integration). Basically, if app tests are instrumented with both `dd-sdk-ios` and `dd-swift-testing`, the CI Visibility SDK requests "force upload all data" in our SDK:
https://github.com/DataDog/dd-sdk-ios/blob/fcf5daa3155d3e29617a193d0fba3987b6569e01/Sources/Datadog/FeaturesIntegration/CITestIntegration.swift#L61

In V2, this logic will spawn from RUM Feature (CI Visibility sets values on RUM events), which only knows `DatadogCoreProtocol`, so we will need something similar. Indeed, it won't need to be "sync" (we can make it "async" and for the purpose of CI visibility we just stall it).

OK with no action right now, but leaving this comment for awareness.

Reply:
Yes good to know 👍  Indeed, we would need to find a way to expose flush logic in test, but I think we can make it test only.